File: src/main/java/org/apache/fineract/client/services/MixReportApi.java
/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements. See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership. The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
package org.apache.fineract.client.services;

import org.apache.fineract.client.CollectionFormats.*;

import io.reactivex.Observable;
import io.reactivex.Completable;
import retrofit2.Call;
import retrofit2.http.*;

import okhttp3.RequestBody;
import okhttp3.ResponseBody;
import okhttp3.MultipartBody;

import java.time.OffsetDateTime;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public interface MixReportApi {
  /**
   * 
   * 
   * @param startDate  (optional)
   * @param endDate  (optional)
   * @param currency  (optional)
   * @return Call&lt;String&gt;
   */
  @GET("mixreport")
  Call<String> retrieveXBRLReport(
    @retrofit2.http.Query("startDate") OffsetDateTime startDate, @retrofit2.http.Query("endDate") OffsetDateTime endDate, @retrofit2.http.Query("currency") String currency
  );

}

File: src/main/kotlin/org/openapitools/client/apis/MixReportApi.kt
package org.openapitools.client.apis

import retrofit2.http.GET
import retrofit2.http.Query


interface MixReportApi {
    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param startDate  (optional)
     * @param endDate  (optional)
     * @param currency  (optional)
     * @return [kotlin.String]
     */
    @GET("v1/mixreport")
    suspend fun retrieveXBRLReport(
        @Query("startDate") startDate: java.time.OffsetDateTime? = null,
        @Query("endDate") endDate: java.time.OffsetDateTime? = null,
        @Query("currency") currency: String? = null
    ): String

}
